File: Polimorfismo/Ej07_Parada/Parada.py
class Parada:
    MAX_ADMINS = 10
    MAX_AUTOS = 10
    def __init__(self, nombre_p="Parada Central"):
        self.nombre_p = nombre_p
        self.admins = []
        self.autos = []
        self.nro_admins = 0
        self.nro_autos = 0
    def adicionar(self, p1, p2=None, p3=None):
        if p2 is None and p3 is None:
            if self.nro_admins < self.MAX_ADMINS:
                self.admins.append(p1)
                self.nro_admins += 1
                print(f"Admin '{p1}' añadido.")
            else:
                print("No se pueden añadir más admins.")
        elif p2 is not None and p3 is not None:
            if self.nro_autos < self.MAX_AUTOS:
                self.autos.append([p1, p2, p3])
                self.nro_autos += 1
                print(f"Auto con placa '{p3}' añadido.")
            else:
                print("No se pueden añadir más autos.")
        else:
            print("Argumentos inválidos para adicionar.")
    def mostrar(self):
        print(f"\n--- {self.nombre_p} ---")
        print(f"Administradores ({self.nro_admins}/{self.MAX_ADMINS}):")
        if not self.admins:
            print("  (Ninguno)")
        else:
            for admin in self.admins:
                print(f"  - {admin}")        
        print(f"Autos ({self.nro_autos}/{self.MAX_AUTOS}):")
        if not self.autos:
            print("  (Ninguno)")
        else:
            for auto in self.autos:
                print(f"  - Modelo: {auto[0]}, Conductor: {auto[1]}, Placa: {auto[2]}")
if __name__ == '__main__':
    parada_defecto = Parada()
    parada_nombre = Parada("Parada de San Pedro")
    parada_defecto.adicionar("Luis Garcia")
    parada_defecto.adicionar("Toyota Corolla", "Ana Flores", "2468ABC")
    parada_defecto.adicionar("Nissan Sentra", "Juan Quispe", "1357XYZ")    
    parada_defecto.mostrar()
    parada_nombre.mostrar()

File: Polimorfismo/Ej07_Parada/Parada.java
package Polimorfismo.Ej07_Parada;
public class Parada {
    private String nombreP;
    private String[] admins;
    private String[][] autos;
    private int nroAdmins;
    private int nroAutos;
    private static final int MAX_ADMINS = 10;
    private static final int MAX_AUTOS = 10;
    public Parada() {
        this.nombreP = "Parada por Defecto";
        this.admins = new String[MAX_ADMINS];
        this.autos = new String[MAX_AUTOS][3];
        this.nroAdmins = 0;
        this.nroAutos = 0;
    }
    public Parada(String nombreP) {
        this.nombreP = nombreP;
        this.admins = new String[MAX_ADMINS];
        this.autos = new String[MAX_AUTOS][3];
        this.nroAdmins = 0;
        this.nroAutos = 0;
    }
    public void adicionar(String admin) {
        if (this.nroAdmins < MAX_ADMINS) {
            this.admins[this.nroAdmins] = admin;
            this.nroAdmins++;
            System.out.println("Admin '" + admin + "' añadido.");
        } else {
            System.out.println("No se pueden añadir más admins.");
        }
    }
    public void adicionar(String modelo, String conductor, String placa) {
        if (this.nroAutos < MAX_AUTOS) {
            this.autos[this.nroAutos][0] = modelo;
            this.autos[this.nroAutos][1] = conductor;
            this.autos[this.nroAutos][2] = placa;
            this.nroAutos++;
            System.out.println("Auto con placa '" + placa + "' añadido.");
        } else {
            System.out.println("No se pueden añadir más autos.");
        }
    }
    public void mostrar() {
        System.out.println("\n--- " + this.nombreP + " ---");
        System.out.println("Administradores (" + this.nroAdmins + "/" + MAX_ADMINS + "):");
        if (nroAdmins == 0) {
            System.out.println("  (Ninguno)");
        } else {
            for (int i = 0; i < nroAdmins; i++) {
                System.out.println("  - " + this.admins[i]);
            }
        }  
        System.out.println("Autos (" + this.nroAutos + "/" + MAX_AUTOS + "):");
        if (nroAutos == 0) {
            System.out.println("  (Ninguno)");
        } else {
            for (int i = 0; i < nroAutos; i++) {
                System.out.println("  - Modelo: " + autos[i][0] + ", Conductor: " + autos[i][1] + ", Placa: " + autos[i][2]);
            }
        }
    }
    public static void main(String[] args) {
        Parada paradaDefecto = new Parada();
        Parada paradaNombre = new Parada("Parada de San Pedro");
        paradaDefecto.adicionar("Luis Garcia");
        paradaDefecto.adicionar("Toyota Corolla", "Ana Flores", "2468ABC");
        paradaDefecto.adicionar("Nissan Sentra", "Juan Quispe", "1357XYZ");
        paradaDefecto.mostrar();
        paradaNombre.mostrar();
    }
}